Zend/zend_execute.c

```diff
@@ -941,50 +936,52 @@ static zend_always_inline const zend_class_entry *zend_ce_from_type(
 		}
 		return ce;
 	}
-	return resolve_single_class_type(name, info->ce);
+	return resolve_single_class_type(name, class_ce);
 }
 
-static bool zend_check_intersection_for_property_class_type(zend_type_list *intersection_type_list,
-	const zend_property_info *info, const zend_class_entry *object_ce)
+static bool zend_check_intersection_for_property_or_class_constant_class_type(zend_type_list *intersection_type_list,
+	const zend_class_entry *class_ce, const zend_class_entry *object_ce)
 {
 	zend_type *list_type;
 
 	ZEND_TYPE_LIST_FOREACH(intersection_type_list, list_type) {
 		ZEND_ASSERT(!ZEND_TYPE_HAS_LIST(*list_type));
-		const zend_class_entry *ce = zend_ce_from_type(info, list_type);
+		const zend_class_entry *ce = zend_ce_from_type(class_ce, list_type);
 		if (!ce || !instanceof_function(object_ce, ce)) {
 			return false;
 		}
 	} ZEND_TYPE_LIST_FOREACH_END();
 	return true;
 }
 
-static bool zend_check_and_resolve_property_class_type(
-		const zend_property_info *info, const zend_class_entry *object_ce) {
-	if (ZEND_TYPE_HAS_LIST(info->type)) {
+static bool zend_check_and_resolve_property_or_class_constant_class_type(
+		zend_type type, const zend_class_entry *class_ce, const zend_class_entry *object_ce) {
```

Comment:
> I found these parameters confusing. 

Yes, so did I, but I only wanted to touch them as much as strictly possible, but I can rename the params if you wish!

However, I'm not sure whether I agree with renaming the function since it implements type checks for properties and constants which behavior is not applicable for params and return types, so I think it's better to specify the target in the name even if it makes the function name too verbose. Or do you think it's still better to rename it?